openedx/features/user_leads/helpers.py

```diff
@@ -0,0 +1,29 @@
+from django.db import IntegrityError, transaction
+
+from .constants import UTM_PARAM_NAMES
+from .models import UserLeads
+
+
+def get_utm_params(request):
+    request_get_params = request.GET
+    utm_params = {}
+
+    for key in UTM_PARAM_NAMES.values():
+        utm_param = request_get_params.get(key)
+        if utm_param:
+            utm_params.update({key: utm_param})
+
+    return utm_params
+
+
+def save_user_utm(request):
+    user = request.user
+    origin = request.resolver_match.url_name
+    utm_params = get_utm_params(request)
+
+    if not user.is_anonymous():
+        try:
+            with transaction.atomic():
+                UserLeads.objects.create(user=user, origin=origin, **utm_params)
+        except IntegrityError:
+            pass
```

Comment:
please log the exception

Reply:
How this was implemented earlier was we were checking for the respective entry from the database first. But as per my discussion with @ahmed-faraz46 we can just try inserting an object and `pass` on the IntegrityError if one occurs. We do not need to log this, because we are looking for this exception on purpose and this is not to handle a misbehaviour, but rather the regular flow of our logic. This exception, if logged, would log for each authenticated user `n-1` time `n` being the number of their visits to the idea or market-listing page, and for sure clutter our logs.

Reply:
get_or_create istemal kerle?

Reply:
@saad-waqar nice suggestion, done and unit tests updated